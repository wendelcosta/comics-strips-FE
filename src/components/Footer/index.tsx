import React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) =>
  createStyles({
    footer: {
      padding: 10,
      backgroundColor: theme.custom.blue,
      color: theme.custom.white,
      textAlign: 'center',
    },
  })
)

const Footer: React.FC = () => {
  const classes = useStyles()
  const fullYear = new Date().getFullYear()
  return (
    <footer className={classes.footer}>
      <p>Copyright {fullYear} &copy; Wendel Costa</p>
    </footer>
  )
}

export default Footer
