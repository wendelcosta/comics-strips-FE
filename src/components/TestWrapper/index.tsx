import React from 'react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../../styles/theme'

const TestWrapper: React.FC = ({ children }: any) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default TestWrapper
