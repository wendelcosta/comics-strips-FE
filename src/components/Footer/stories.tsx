import React from 'react'

import Footer from '.'
import TestWrapper from '../TestWrapper'

export default {
  title: 'Footer',
  component: Footer,
}

export const Default = (): any => (
  <TestWrapper>
    <Footer />
  </TestWrapper>
)
