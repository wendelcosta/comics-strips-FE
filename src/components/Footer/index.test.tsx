import React from 'react'
import { render } from '@testing-library/react'

import TestWrapper from '../TestWrapper'

import Footer from '.'

describe('Footer Test', () => {
  test('Loads Footer', () => {
    const { container } = render(
      <TestWrapper>
        <Footer text="Copyright Wendel Costa" />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
