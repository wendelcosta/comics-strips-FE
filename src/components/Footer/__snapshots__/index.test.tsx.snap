// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test Loads Footer 1`] = `
<div>
  <footer
    class="makeStyles-footer-1"
  >
    <p>
       
      2020
       Â© 
      Copyright Wendel Costa
    </p>
  </footer>
</div>
`;
